pymeasure/instruments/hp/hp6641a.py

```diff
@@ -0,0 +1,209 @@
+        psu.output_enabled = 1
+        print(f'read errors: {psu.pop_err()}')
+        while True:
+            print(f'output voltage: {psu.voltage}')
+            print(f'output current: {psu.current}')
+            time.sleep(1)
+    """
+
+    BOOL_MAPPINGS = {True: 1, False: 0}
+    DISP_MAPPINGS = {True: "TEXT", False: "NORM"}
+
+    def __init__(self, adapter, name="HP6641A", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+
+    voltage_setpoint = Instrument.control(
+        "VOLT?", "VOLT %g",
+        "Control the voltage setpoint in Volts (float).",
+        dynamic=True,
+        validator=strict_range,
+        values=[0, limits["HP6641A"]["Volt_lim"]]
+    )
+
+    ovp_setpoint = Instrument.control(
+        "VOLT:PROT?", "VOLT:PROT %g",
+        "Control the Over Voltage Protection setpoint in Volts (float).",
+        dynamic=True,
+        validator=strict_range,
+        values=[0, limits["HP6641A"]["OVP_lim"]]
+    )
+
+    current_setpoint = Instrument.control(
+        "CURR?", "CURR %g",
+        "Control the current setpoint in Amperes (float).",
+        dynamic=True,
+        validator=strict_range,
+        values=[0, limits["HP6641A"]["Cur_lim"]]
+    )
+
+    voltage = Instrument.measurement(
+        "MEAS:VOLT?",
+        "Measure the voltage at the power supply's sense terminals"
+    )
+
+    current = Instrument.measurement(
+        "MEAS:CURR?",
+        "Measure the current at the power supply's sense terminals"
+    )
+
+    output_enabled = Instrument.control(
+        "OUTP:STAT?", "OUTP:STAT %g",
+        "Control the power supply output and Read back the programmed value",
+        validator=strict_discrete_set,
+        values=BOOL_MAPPINGS,
+        map_values=True
+    )
+
+    disp_mode_text = Instrument.control(
+        "DISP:MODE?", "DISP:MODE %s",
+        "Set and Get display mode between its normal metering mode and a mode in which\
+        it displays text sent by the user",
+        validator=strict_discrete_set,
+        values=DISP_MAPPINGS,
+        map_values=True
+    )
+
+    disp_text = Instrument.control(
+        "DISP:TEXT?", "DISP:TEXT \"%s\"",
+        "Set and Get character strings to/from display",
+    )
+
+    scpi_version = Instrument.measurement(
+        "SYST:VERS?",
+        """Get SCPI version of the multimeter.""",
+    )
+
+    self_test_result = Instrument.measurement(
+        "*TST?",
+        """Get self-test result.
+
+        Be sure to set an appropriate connection timeout,
+        otherwise the command will fail.""",
+    )
+
+    def pop_err(self):
+        """Pop an error off the error queue. Returns a tuple containing the
+        code and error description. An error code 0 indicates success."""
+        error_str = self.ask('SYST:ERR?').strip()
+        error_code_str, error_desc_str = error_str.split(',')
+        error_code = int(error_code_str)
+        error_desc = error_desc_str.strip('"')
+        return error_code, error_desc
```

Comment:
Something like that is already implemented in SCPIMixin.next_error.

If you need this method at all, it would be a prime candidate to add a protocol test for.

Reply:
This method was copied from hp34401

Reply:
Ok. Please have a look at SCPIMixin.next_error and whether it suits your needs.

Reply:
next_error return raw string, pop_err return code (int) and description (string).
The second option is more useful.

Reply:
No, next error returns an int and a string:
https://github.com/pymeasure/pymeasure/blob/c1ef467e884e01d7209a219612525de7bc74a192/tests/instruments/test_generic_types.py#L54-L59

The property creator magic splits already at the comma and casts to numbers.